client/reader/search-stream/suggestion-provider.jsx

```diff
@@ -19,36 +20,75 @@ import { getReaderFollowedTags } from 'state/selectors';
  * @return {Array}       An array of suggestions, or null if no tags where provided
  */
 function suggestionsFromTags( count, tags ) {
+	const tagSuggestions = [];
+	let sampleSuggestions = {};
 	if ( tags ) {
 		if ( tags.length <= count ) {
 			return [];
 		}
-		return map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+		sampleSuggestions = map( sampleSize( tags, count ), tag => ( tag.displayName || tag.slug ).replace( /-/g, ' ' ) );
+
+		for ( let i = 0; i < sampleSuggestions.length; i++ ) {
+			tagSuggestions.push( {
+				text: sampleSuggestions[ i ],
+				railcar: {
+					railcar: analytics.tracks.createRandomId() + '-' + i,
+					ui_algo: 'read:search-suggestions:tags/1',
+					ui_position: i,
+					rec_result: sampleSuggestions[ i ],
+				}
+			} );
+		}
+
+		return tagSuggestions;
 	}
 	return null;
 }
 
 function suggestionsFromPicks( count ) {
 	const lang = i18nUtils.getLocaleSlug().split( '-' )[ 0 ];
+	const pickSuggestions = [];
+	let sampleSuggestions = {};
 
 	if ( suggestions[ lang ] ) {
-		return sampleSize( suggestions[ lang ], count );
+		sampleSuggestions = sampleSize( suggestions[ lang ], count );
+
+		for ( let i = 0; i < sampleSuggestions.length; i++ ) {
+			pickSuggestions.push( {
+				text: sampleSuggestions[ i ],
+				railcar: {
+					railcar: analytics.tracks.createRandomId() + '-' + i,
+					ui_algo: 'read:search-suggestions:picks/1',
+					ui_position: i,
+					rec_result: sampleSuggestions[ i ],
+				}
+			} );
+		}
+
+		return pickSuggestions;
 	}
 	return null;
 }
 
 function getSuggestions( count, tags ) {
```

Comment:
This block could be clearer with something like this:
```js
const tagSuggestions = suggestionsFromTags( count, tags );

// return null to suppress showing any suggestions until tag subscriptions load.
if ( tagSuggestions === null ) {
  return null;
}

const suggestions = !! tagSuggestions.length
  ? tagSuggestions
  : suggestionsFromPicks( count );

trackSuggestionRailcarRender( suggestions );
return suggestions;
```

The major change is in naming / mutability. `tagSuggestions` always refers to suggestionsFromTags, whereas the `const` suggestions always refers to the item going to be returned.